chain/client/src/client_actor.rs

```diff
@@ -1450,6 +1452,19 @@ impl ClientActor {
         !self.adv.disable_header_sync() && needs_syncing
     }
 
+    fn run_flat_storage_creation_step(&mut self, ctx: &mut Context<ClientActor>) {
+        if let Err(err) = self.client.run_flat_storage_creation_step() {
```

Comment:
A small optimization we can add here is to let `run_flat_storage_creation_step` return a boolean indicating whether FlatStorageCreator is None or all the shards are already ready, and stop scheduling new run_later if that's true. This way, we can save some unnecessary scheduling